target/linux/ramips/mt7621/base-files/etc/board.d/02_network

```diff
@@ -140,6 +140,24 @@ ramips_setup_macs()
 		label_mac=$(cat "/sys/firmware/mikrotik/hard_config/mac_base")
 		wan_mac=$label_mac
 		lan_mac=$(macaddr_add $label_mac 1)
+
+		lan_list=""
+		for lan in /sys/class/net/lan*; do
+			lan_list="$lan_list $(basename $lan)"
+		done
+
+		lans_mac="$lan_mac"
+		for lan in $lan_list; do
+			ucidef_set_network_device_mac $lan $lans_mac
+			lans_mac=$(macaddr_add $lans_mac 1)
+		done
+
+		ucidef_set_network_device_mac "wan" $label_mac
+
+		if [ "$board" = mikrotik,routerboard-760igs ]; then
```

Comment:
I won't accept any nested ifs here. We had that in ar71xx and I'm glad it's gone now.